yb-voyager/src/queryparser/queryParser.go

```diff
@@ -0,0 +1,33 @@
+package queryparser
+
+import (
+	pg_query "github.com/pganalyze/pg_query_go/v5"
+)
+
+type QueryParser struct {
+	QueryString string
+	ParseTree   *pg_query.ParseResult
+}
+
+func New(query string) *QueryParser {
+	return &QueryParser{
+		QueryString: query,
+	}
+}
+
+func (qp *QueryParser) Parse() error {
+	tree, err := pg_query.Parse(qp.QueryString)
+	if err != nil {
+		return err
+	}
+	qp.ParseTree = tree
+	return nil
+}
+
+func (qp *QueryParser) CheckUnsupportedQueryConstruct() (string, error) {
```

Comment:
It is technically possible for a query to have multiple unsupported query constructs, right? I think you should design it that way.

Reply:
Yeah correct.
Let me make the changes.